packages/nextjs/src/server/data/getAuthDataFromRequest.ts

```diff
@@ -37,21 +52,59 @@ export function getAuthDataFromRequest(
     authReason,
   };
 
-  opts.logger?.debug('auth options', options);
+  // Only accept session tokens in sync version
+  if (!isTokenTypeAccepted(TokenType.SessionToken, opts.acceptsToken || TokenType.SessionToken)) {
+    return signedOutAuthObject(options);
+  }
 
-  let authObject;
   if (!authStatus || authStatus !== AuthStatus.SignedIn) {
-    authObject = signedOutAuthObject(options);
-  } else {
-    assertTokenSignature(authToken as string, options.secretKey, authSignature);
+    return signedOutAuthObject(options);
+  }
+
+  assertTokenSignature(authToken as string, options.secretKey, authSignature);
+  const jwt = decodeJwt(authToken as string);
+  opts.logger?.debug('jwt', jwt.raw);
+
+  // @ts-expect-error -- Restrict parameter type of options to only list what's needed
+  return signedInAuthObject(options, jwt.raw.text, jwt.payload);
+};
+
+/**
+ * Note: We intentionally avoid using interface/function overloads here since these functions
+ * are used internally. The complex type overloads are more valuable at the public API level
+ * (like in auth.protect(), auth()) where users interact directly with the types.
+ *
+ * Given a request object, builds an auth object from the request data. Used in server-side environments to get access
+ * to auth data for a given request.
+ */
+export const getAuthDataFromRequestAsync = async (
+  req: RequestLike,
+  opts: GetAuthDataFromRequestOptions = {},
+): Promise<AuthObject> => {
```

Comment:
We created a separate function since verifying machine tokens is async, and it will break `getAuth()`. For now, machine verification only applies to `auth()`.